### PR TITLE
[CAZB-2658] Leeds taxi - back button implementation

### PR DESCRIPTION
**Pull request by pkalamarski**
## Motivation and Context

https://eaflood.atlassian.net/browse/CAZB-2658

## How Has This Been Tested?

Manually, cucumber and rspec added
## Checklist:

- [x] It contains only changes required by issue (does not contain other PR)
- [x] Includes the link to an issue (if apply)
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: app/controllers/dates_controller.rb [hunk 1]
```diff
@@ -264,7 +290,13 @@ def confirm_date_weekly
   # ==== Validations
   # * +weekly_charge_today+ - lack of value in the session redirects to {weekly_charge}[rdoc-ref:weekly_charge]
   #
-  def select_weekly_period; end
+  def select_weekly_period
+    @return_path = if session[:first_week_start_date] || session[:second_week_start_date]
```
Comment (pkalamarski) [inline app/controllers/dates_controller.rb:294]: remove second week

File: app/services/dates/assign_back_button_date.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+module Dates
+  ##
+  #
+  class AssignBackButtonDate < BaseService
+    def initialize(session:, week: nil)
+      @session = session
+      @week = week
+    end
+
+    def call
```
Comment (pkalamarski) [inline app/services/dates/assign_back_button_date.rb:14]: split to methods

File: app/services/dates/assign_back_button_date.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    end
+
+    private
+
+    attr_reader :session, :week
+
+    # Lists disabled dates for selected week start date
+    # Disables 6 days before (to prevent payment overlapping) and 6 days after week_start_dates
+    # Returns array of dates in string YYYY-MM-DD format
+    def disable_week(week_start_date)
```
Comment (pkalamarski) [inline app/services/dates/assign_back_button_date.rb:36]: move to module

File: spec/requests/dates_controller/select_weekly_period_spec.rb [hunk 4]
```diff
@@ -7,11 +7,31 @@
 
   context 'with VRN and LA in the session' do
     context 'when Leeds weekly discount is possible' do
-      before { add_details_to_session(weekly_possible: true) }
+      before do
+        add_details_to_session(weekly_possible: true, weekly_charge_today: true)
+      end
```
Comment (likeuwill) [inline spec/requests/dates_controller/select_weekly_period_spec.rb:12]: `before { ... }`

File: spec/services/dates/assign_back_button_date_spec.rb [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+require 'rails_helper'
+
+RSpec.describe Dates::AssignBackButtonDate do
+  include DatesHelper
+
+  subject(:service) do
+    described_class.call(
+      session: session,
+      second_week_selected: second_week_selected
+    )
```
Comment (likeuwill) [inline spec/services/dates/assign_back_button_date_spec.rb:12]: one line

**Pull request merged**